### PR TITLE
Add support for Alveo U55C cards

### PR DESCRIPTION
**Pull request by LinusJungemann**
Adds support for the Alveo U55C FPGA card to FINN.

Additionally, support to choose the FPGA device to be used for execution of the synthesized design is added to the FINN driver.

### CODE REVIEW

**Comment by auphelia** [comment]
Hi @LinusJungemann , thanks for your contribution!
Could you please make sure that [the linting](https://github.com/Xilinx/finn/actions/runs/6706509360/job/18294654942?pr=913) is successful? That will ensure that the style of your changes is in line with the rest of the repository.

To setup pre-commit correctly, you can use the [pre-commit configuration file](https://github.com/Xilinx/finn/blob/dev/.pre-commit-config.yaml) that is provided in the repository. If there are issues to run pre-commit in the Docker container, a separate Python 3.10 environment can be used for that as well.

**Comment by LinusJungemann** [comment]
Hi @auphelia, sorry I was not able to fix the linting until now, because of other stuff I had to do.
It should be fixed now with the proper githook.

File: src/finn/util/platforms.py [hunk 1]
```diff
@@ -461,11 +461,46 @@ def compute_resources(self):
         ]
 
 
+class Alveo_NxU55C_Platform(Platform):
```
Comment (auphelia) [inline src/finn/util/platforms.py:464]: How did you determine these settings for U55C?

**Review by auphelia — CHANGES_REQUESTED** [review]
Thanks @LinusJungemann .
Overall the PR looks good, there is just a small change request in the driver_base and could you please clarify how you determined the settings for the U55C platform in `platforms.py`?

File: src/finn/qnn-data/templates/driver/driver_base.py [hunk 2]
```diff
@@ -78,6 +78,7 @@ def __init__(
             Path to runtime weights folder.
         """
         super().__init__(bitfile_name, download=download, device=device)
+        self.device = device
```
[the remark this answers is not in the document]
Comment (LinusJungemann) [reply src/finn/qnn-data/templates/driver/driver_base.py]: Yes. I removed it in the latest commit.

File: src/finn/util/platforms.py (hunk 1 above)
Comment (LinusJungemann) [reply src/finn/util/platforms.py:464]: According to our knowledge, the U55C is identical to the U280 cards (based on https://github.com/erinadreno/list_of_Xilinx_FPGAs#Alveo-and-Kria) in regards to the PL.
Therefore, this calculation is based on the calculations for the U280 card that is already included.

Comment (auphelia) [reply src/finn/util/platforms.py:464]: Thanks, could you please update the metrics according to [UG1102](https://docs.xilinx.com/r/en-US/ug1120-alveo-platforms/U55C-Gen3x16-XDMA-base_3-Platform)? 
E.g. the number of slr (`nslr`) for the U55C is 3 and not 2 and the ethernet connection is on SLR1. In the documentation from the link above you can also find the resources per SLR that are not completely identical with the U280. I would also suggest, that the lines that are part of "observed from Vivado" are taking out, unless you have experimental results for that.

Comment (LinusJungemann) [reply src/finn/util/platforms.py:464]: Everything should be fixed now based on the UG1102 document.

**Review by auphelia — APPROVED** [review]
Thanks @LinusJungemann , looks great now!

**Pull request merged**